##Installation
    $ git clone git://github.com/hacone/AgIn.git AgIn

Demo files are available at

http://www.gi.k.u-tokyo.ac.jp/~hacone/Demo.tar.gz

##Analyse your data
The program requires a `modification.csv` file as a primary input, which is produced by `P_ModificationDetection` module of SMRT Pipe.
Then run as:

    $ target/dist/bin/launch -i path/to/modifications.csv -f path/to/reference.fasta -o output_prefix -b vector_specifier -g gamma -l min_length predict

It'll give three output files:

    output_prefix.gff
    output_prefix_class.wig
    output_prefix_coverage.wig

The program assumes the index file for FASTA file (`.fai` generated by samtools) is placed in the same directory as FASTA.


####The parameters for prediction
`-b` : either `P5C3_HdrR` (for P5-C3 sequencing), or `LDAVector` (for P4-C2 and C2-C2 sequencing)

`-l` : Minimum number of CpG sites the predicted regions must contain

`-g` : Decision threshold.

Recommended (standard) combinations of parameters are:
    -b P5C3_HdrR -g -0.88 -l 50 (for P5-C3)
    -b LDAVector -g -1.80 -l 50 (for P4-C2 or C2-C2)


## Output files
The GFF file reports all the predicted hypomethylated regions.

`_class.wig` file describes the predicted class (0 for unmethylated, 1 for methylated) of each CpG site.

`_coverage.wig` file describes the effective coverage used for making the prediction. (i.e., # of IPD ratio measurements on 21 bases around the CpG site / 21.0 )

## Lisence

Copyright of this program `AgIn` is held by Yuta Suzuki
(ysuzuki@cb.k.u-tokyo.ac.jp) and is distributed under the (modified) BSD license.

Copyright of `src/script/launch` and some other scripts is held by Taro L.
Saito(leo@xerial.org) and licensed under Apache License Version 2.0
(http://www.apache.org/licenses/LICENSE-2.0)
